kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -247,29 +249,60 @@ define([
+
+            const $fileTable = $(stagingAreaViewer.ftpFileTableTmpl({
                 files: files,
-                uploaders: this.uploaders.dropdown_order
+                uploaders: stagingAreaViewer.uploaders.dropdown_order
             }));
-            this.$elem.append($fileTable);
-            this.$elem.find('table').dataTable({
+            
+            stagingAreaViewer.$elem.append($fileTable);
+            stagingAreaViewer.$elem.find('table').dataTable({
                 language: {
                     emptyTable: emptyMsg
                 },
                 dom: '<"file-path pull-left">frtip',
                 autoWidth: false,
                 order: [[4, 'desc']],
+                headerCallback: function (thead) {     
+               
+                    function selectAllOrNone (event) {
+                        let selectAllChecked = event.target.checked;
+                        
+                        $('input.kb-staging-table-body__checkbox-input:enabled')
+                        .prop('checked', selectAllChecked)
+                        .attr('aria-checked', selectAllChecked);
+
+                        //enable or disable import appropriately
+                        if (selectAllChecked) {
+                            stagingAreaViewer.enableImportButton();
+                        } else {
+                            stagingAreaViewer.disableImportButton();
+                        }
+                    }
+
+                    $(thead).find('th').eq(0)
+                    .on('click', (e) => {
+                        selectAllOrNone(e);
+                    })
+                    .on('keyPress', (e) => {
+                        selectAllOrNone(e);
+                    });                    
+                },
                 columnDefs: [{
                     targets: 0,
                     orderable: false,
                     searchable: false,
-                    //TODO: what value makes most sense for these? file name, or eventual data type selection? 
-                    render: function (data, type, full) {
+                    render: function (data) {
                         //render checkboxes disabled until the user selects a type
-                        return '<input class="kb-staging-table-body__checkbox-input" type="checkbox" name="id[]" disabled=true value="'
-                           + $('<div/>').text(data).html() + '">';
+                        return '<input class="kb-staging-table-body__checkbox-input" \
+                        type="checkbox" disabled=true \
+                        aria-checked="false" tabindex="0" aria-label="Select to import file checkbox: disabled until at least one data type is selected" \
+                        id="'+ $('<div/>').text(data).html() + '">';
+                        //TODO: this ID is not guaranteed if user uploads more than one file with same name - is that an edge case we care about? 
```

Comment:
Added some accessibility tags to this element. Also not sure if ID is the right way to go - plz discuss

Reply:
does `data` get cleaned up at all? what happens if it contains spaces or characters that aren't permitted in JS IDs?

Reply:
Answering my own question... no, it doesn't. If the ID only used to uniquely identify that checkbox (e.g. for a label element) but isn't parsed to get the file name or something like that, I'd see if there's something else you could use -- e.g. does datatables keep a row_id that you can access? Can you use some sort of nonce function to create a unique identifier?

Reply:
if the ID is used to convey data information, would it be better to move that info into a `data-` attribute so that it doesn't matter if the information is not compliant with HTML5 / JS rules for IDs?

Reply:
i like your suggestion - i will go ahead and move the file name into a data- attribute and make id a legit unique ID